ner_v2/detectors/numeral/number_range/standard_number_range_detector.py

```diff
@@ -0,0 +1,385 @@
+        for number_tag in self.number_detected_map:
+            original = original.replace(number_tag, self.number_detected_map[number_tag][1])
+        if original == number_tag_text:
+            return None
+        return original
+
+    def detect_number_range(self, text):
+        """
+        Detect number-range from number range text. Run through list of detectors defined in detector_preferences in
+        the preferences.
+        Args:
+            text(str): text string
+        Returns:
+            number_list (list): list containing detected numeric text
+            original_list (list): list containing original numeral text
+
+        """
+        self.text = text
+        self.processed_text = text
+        self.tagged_text = text
+        self.number_detected_map = self._get_number_tag_dict()
+
+        number_list, original_list = None, None
+        for detector in self.detector_preferences:
+            self.number_tagged_processed_text = self._tag_number_in_text(self.processed_text)
+            number_list, original_list = detector(number_list, original_list)
+            self._update_processed_text(original_list)
+        return number_list, original_list
+
+    def _update_number_range_and_original_list(self, number_tag_min, number_tag_max, matched_text,
+                                               number_range_list=None, original_list=None):
+        """
+        Update number_range_list and original_list by finding entity value of number tag and original text from
+        number_detected_map
+        Args:
+            number_tag_min (str or None): tagged min number
+            number_tag_max (str or None): tagged max number
+            matched_text (str): text matching regex
+            number_range_list (list): list containing all number range detected
+            original_list (list): list containing original text of values detected
+        Returns:
+            number_range_list (list)
+            original_list (list)
+        """
+        number_range_list = number_range_list or []
+        original_list = original_list or []
+
+        entity_value_min, entity_value_max, entity_unit = None, None, None
+
+        if number_tag_min and number_tag_min in self.number_detected_map:
+            entity_dict = self.number_detected_map[number_tag_min][0]
+            entity_value_min = entity_dict[NUMBER_DETECTION_RETURN_DICT_VALUE]
+            entity_unit = entity_dict[NUMBER_DETECTION_RETURN_DICT_UNIT]
+
+        if number_tag_max and number_tag_max in self.number_detected_map:
+            entity_dict = self.number_detected_map[number_tag_max][0]
+            entity_value_max = entity_dict[NUMBER_DETECTION_RETURN_DICT_VALUE]
+            entity_unit = entity_dict[NUMBER_DETECTION_RETURN_DICT_UNIT]
+
+        if self.unit_type and \
+                (entity_unit is None or
+                 self.number_detector_object.language_number_detector.units_map[entity_unit].type != self.unit_type):
```

Comment:
We have to put this check inside Number range because in `26 to 37 km`, `37` will be tagged but there is no unit in `26` it will not be tagged because we want to detect distance unit only. So we have to make number detector tag entity irrespective of unit. And then we can check in number range if  we are getting unit or not.

Reply:
Sorry for being ambiguous, I meant that we can make the dict access a method of numberdetector. Also, I feel like the condition has some issues

```python
class NumberDetector(...):
    def get_unit_type(detected_unit):
        unit = self.language_number_detector.units_map.get(entity_unit)
        type = unit.type if unit else None
        return type
```

then this condition becomes
```python
# if self.unit_type is None, we want everything or else it detected_unit must have type equal to self.unit_type
if (not self.unit_type or
       (self.unit_type and self.number_detector. get_unit_type(detected_unit) != self.unit_type)):
```

Reply:
Done